From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect backing-dev include in mm/migrate
Patch-mainline: never, kabi
References: CVE-2016-3070, bsc#979215, VM Functionality

In 3.12.63, commit 2c789028e7390ca050752f659a82c6cdad357b67 (mm:
migrate dirty page without clear_page_dirty_for_io etc), upstream
commit 42cb14b110a5698ccf26ce59c4441722605a3743 added one #include
into mm/migrate.c. It made struct backing_dev_info and others visible
to the kABI checker. Hence it reports kABI changes now.

So, just protect the include by __GENKSYMS__ to hide it from the
checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 mm/migrate.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -31,7 +31,9 @@
 #include <linux/vmalloc.h>
 #include <linux/security.h>
 #include <linux/memcontrol.h>
+#ifndef __GENKSYMS__
 #include <linux/backing-dev.h>
+#endif
 #include <linux/syscalls.h>
 #include <linux/hugetlb.h>
 #include <linux/hugetlb_cgroup.h>
